public/docs/_examples/dependency-injection/dart/lib/providers_component.dart

```diff
@@ -309,39 +308,43 @@ class ProviderComponent10b implements OnInit {
   // #enddocregion provider-10-ctor
   ngOnInit() {
     // #docregion provider-10-logger
-
     // No logger? Make one!
     if (_logger == null) {
-      _logger = new Logger();
+      _logger = new DoNothingLogger();
       // #enddocregion provider-10-logger
       _logger.log('Optional logger was not available.');
     } else {
       _logger.log('Hello from the injected logger.');
-      log = _logger.logs[0];
     }
     log = _logger.logs[0];
   }
 }
 
-/////////////////
+// #docregion provider-10-logger
+// . . .
+class DoNothingLogger extends Logger {
+  List<String> logs = [];
```

Comment:
Empty line after this + final + ideal would be to expose a read-only `Iterable<String> logs`.